README.md

```diff
@@ -1,30 +1,177 @@
-props to child components — we can greatly reduce that complexity (and make
-our components dead-simple to test).
+Once a component has state, it becomes harder to predict its output and behavior.
+Because, by definition, state changes, we won't be able to say at any given point
+in our program's operation what state our component is in. That, in turn, will
+make it harder to reason about our application; and when it becomes harder to
+reason about our application we will struggle more to eradicate and avoid bugs.
 
-This is one of the main ideas behind `connect()` (previously, `@connect()`)
-in Redux: we can abstract away state management so that we're only dealing
-with props. We still don't quite want to introduce students to flux or Redux
-quite yet, but we want to get them a bit closer.
+> Note: A general principal that one can take on here, as an arguable truth about
+application design, is that while we often find ourselves needing state in our
+applications, it is state that our human brains struggle most to parse and track.
 
-We can code-along a simple abstraction layer that achieves the above-described
-functionality with callbacks (or actions, since students know about those).
+So, if state is something we should avoid, what are some strategies to achieve
+that goal? This is where `props` come in!
+
+Props provide us with a way to minimize
+the number of components in our app that need to carry and manage state. Another,
+more evocative, way to say this is that `props` provide us with a method for
+reducing the total "surface area" of our app that is stateful. The reason `props`
+help us to minimize `state` is because we can use them to send data _down_
+into our nested child components. Because we can send data _down_,
+we can _push state up_, thereby concentrating state at the "top" of our UI's 
+hierarchy. 
+
+This may all seem a bit abstract at this point, so let's take a look at an example
+to illustrate our point.
+
+## A use-case for pushing state upwards 
+
+Imagine we have an app, a Fruit Basket app. It's simple. It just tracks what's
+in our fruit basket, and what's already been eaten. In designing our application,
+let's say that we initially decide to build two components -- `FruitNotEaten` and
+`FruitEaten` -- each of which is responsible for fetching its own list
+from the app's backend.
+
+Given this plan, our `FruitEaten` component, for example, could look like this:
+
+```javascript
+class FruitEaten extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { 
+      fruitList: []
+    };
+  }
+
+  componentWillMount() {
+    // Assume that this fetch function is a Promise that
+    // can go over the internet.
+    fetch('/api/fruit=?eaten=true').then((data) => {
+      this.setState({ fruitList: data });
+    });
+  }
+
+  render() {
+    return (
+      <ul>
+        {this.state.fruitList.map((fruit) => {
+          return <li key={fruit}>{fruit}</li>;
```

Comment:
Maybe use the arrow function shorthand here to showcase even more of ES2015's awesomeness? Minor suggestion.